### PR TITLE
Module opts map time

### PR DESCRIPTION
**Pull request by JanuszJakubiec**

### CODE REVIEW

File: test/config_parser_SUITE.erl [hunk 1]
```diff
@@ -2919,8 +2919,9 @@ mod_stream_management_stale_h(_Config) ->
     ?errh(T(#{<<"geriatric">> => <<"one">>})).
 
 mod_time(_Config) ->
-    check_iqdisc(mod_time),
-    ?cfgh(modopts(mod_time, []), #{<<"modules">> => #{<<"mod_time">> => #{}}}).
+    check_iqdisc_map(mod_time),
+    check_module_defaults(mod_time),
+    ?cfgh([modules, mod_time, iqdisc], one_queue, #{<<"modules">> => #{<<"mod_time">> => #{}}}).
```
Comment (chrzaszcz) [inline test/config_parser_SUITE.erl:2924]: This is already checked in `check_module_defaults` (see previous line).

**Review by chrzaszcz — COMMENTED** [review]
Looks good in general, I added minor remarks.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/108548) / small_tests / 89a0e81a94ad917740efcd7efb81c9c61962dbef
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3554/108548/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3554/108548/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-22_14.56.11/lib.mongooseim.logs/run.2022-02-22_14.56.12/suite.log.html)

File: src/mod_time.erl [hunk 2]
```diff
@@ -24,12 +24,12 @@
 -xep([{xep, 202}, {version, "2.0"}]).
 -xep([{xep, 82}, {version, "1.1"}]).
 
-start(HostType, Opts) ->
-    IQDisc = gen_mod:get_opt(iqdisc, Opts, one_queue),
+-spec start(HostType :: mongooseim:host_type(), gen_mod:module_opts()) -> ok | {erorr, atom()}.
```
Comment (chrzaszcz) [inline src/mod_time.erl:27]:
```suggestion
-spec start(HostType :: mongooseim:host_type(), gen_mod:module_opts()) -> ok | {error, atom()}.
```

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/108717) / small_tests / 8dba440f45d736fce8b848cf586389d8684fdd6a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3554/108717/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3554/108717/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-23_06.41.44/lib.mongooseim.logs/run.2022-02-23_06.41.44/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/108745) / small_tests / df248961ebba10910a1b7324c8f6fd22cea5bca5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3554/108745/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3554/108745/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-23_06.43.29/lib.mongooseim.logs/run.2022-02-23_06.43.30/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/108894) / small_tests / 69e4b39643c5a4b34ad9d80b241c37ae8c228e35
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3554/108894/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3554/108894/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-23_10.21.29/lib.mongooseim.logs/run.2022-02-23_10.21.29/suite.log.html)

File: big_tests/tests/mod_time_SUITE.erl [hunk 3]
```diff
@@ -39,7 +39,7 @@ suite() ->
 %%--------------------------------------------------------------------
 
 init_per_suite(Config) ->
-    dynamic_modules:start(domain_helper:host_type(), mod_time, []),
+    dynamic_modules:start(domain_helper:host_type(), mod_time, #{iqdisc => one_queue}),
```
Comment (gustawlippa) [inline big_tests/tests/mod_time_SUITE.erl:42]: Similarly to the other PR, you could use defaults from the helper here.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/110086) / small_tests / 52ba9ee3df83921617275981c942a23024d92f1a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3554/110086/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3554/110086/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-01_10.55.51/lib.mongooseim.logs/run.2022-03-01_10.55.52/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/110113) / small_tests / 3686ab5e2868be980f30b339225e84a32953c24e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3554/110113/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3554/110113/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-01_10.57.38/lib.mongooseim.logs/run.2022-03-01_10.57.39/suite.log.html)

**Review by chrzaszcz — COMMENTED** [review]
Looks good 👍

**Review by chrzaszcz — APPROVED** [review]

**Pull request merged**